Sources_v3/StreamChat/Database/DatabaseSession.swift

```diff
@@ -149,13 +149,23 @@ protocol MemberListQueryDatabaseSession {
     func saveQuery<ExtraData: UserExtraData>(_ query: ChannelMemberListQuery<ExtraData>) throws -> ChannelMemberListQueryDTO
 }
 
+protocol MessageReactionDatabaseSesssion {
+    func reaction(messageId: MessageId, userId: UserId, type: ReactionType) throws -> MessageReactionDTO?
+    
+    @discardableResult
+    func saveReaction<ExtraData: ExtraDataTypes>(payload: MessageReactionPayload<ExtraData>) throws -> MessageReactionDTO
+    
+    func delete(reaction: MessageReactionDTO)
+}
+
```

Comment:
Why do you think we need an extra protocol for this? Isn't this part of `MessageDatabaseSession`?

Reply:
haha, you are right! 😄